shared/src/test/factories.ts

```diff
@@ -411,14 +411,16 @@ export const generateTaxClearanceCertificateData = (
   overrides: Partial<TaxClearanceCertificateData>,
 ): TaxClearanceCertificateData => {
   const taxId = randomInt(12).toString();
+  const addressState = generateUnitedStatesStateDropdownOption({});
   return {
     requestingAgencyId: randomElementFromArray(taxClearanceCertificateAgencies).id,
     businessName: `some-business-name-${randomInt()}`,
     addressLine1: `some-address-1-${randomInt()}`,
     addressLine2: `some-address-2-${randomInt()}`,
     addressCity: `some-city-${randomInt()}`,
-    addressState: generateUnitedStatesStateDropdownOption({}),
-    addressZipCode: randomInt(5).toString(),
+    addressState,
+    addressZipCode:
+      addressState.shortCode === "NJ" ? `07${randomInt(3).toString()}` : randomInt(5).toString(),
```

Comment:
[sand] Do we have a New Jersey ZIP Code™ generator function? If we don't, should we refactor this so that other tests can use it to reduce flakiness? That way we can have a rock-solid implementation and change things across the app if our ZIP Codes™ ever change.

Long term, I think that we should have a generic state and postal code generator function that only generates valid info, but I think that's way out of scope here.